quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -45,26 +47,50 @@ trait Parsing extends SchemaConfigParsing {
 
     case q"$i: $typ"                        => astParser(i)
 
-    case q"$tupleTree match { case ($fieldsTrees) => $bodyTree }" =>
-      val tuple = astParser(tupleTree)
-      val fields = astParser(fieldsTrees)
-      val body = astParser(bodyTree)
-      def property(path: List[Int]) =
-        path.foldLeft(tuple) {
-          case (t, i) => Property(t, s"_${i + 1}")
-        }
-      def reductions(ast: Ast, path: List[Int] = List()): List[(Ident, Ast)] =
-        ast match {
-          case ident: Ident =>
-            List(ident -> property(path))
-          case Tuple(elems) =>
-            elems.zipWithIndex.flatMap {
-              case (elem, idx) => reductions(elem, path :+ idx)
-            }
-          case other =>
-            c.fail(s"Please report a bug. Expected tuple or ident, got '$other'")
-        }
-      BetaReduction(body, reductions(fields): _*)
+    case `patMatchParser`(value)            => value
+    case `blockParser`(block)               => block
+  }
+
+  val blockParser: Parser[Block] = Parser[Block] {
+    case q"{..$exprs}" if exprs.size > 0 => Block(exprs.map(astParser(_)))
+  }
+
+  val valParser: Parser[Val] = Parser[Val] {
+    case q"val $name: $typ = $body" => Val(ident(name), astParser(body))
+  }
+
+  val patMatchValParser: Parser[Val] = Parser[Val] {
+    case q"$mods val $name: $typ = ${ patMatchParser(value) }" =>
+      Val(ident(name), value)
+  }
+
+  val patMatchParser: Parser[Ast] = Parser[Ast] {
+    case q"$expr match { case ($fields) => $body }" =>
+      astParser(fields) match {
+        case Val(name, body) => BetaReduction(body, List(name -> body): _*)
```

Comment:
@gustavoamigo yes, not sure why codecov reports that as a miss. The line in question would be matched on multi-assignment as in `val (a,b)` below

```scala
val q = quote {
  query[TestEntity].map{e=>
    val (a,b) = (1,2)
    e.i + a + b
  }
}
```
Maybe [indirection in `patMatchValParser`](https://github.com/godenji/quill/blob/inline-val-support/quill-core/src/main/scala/io/getquill/quotation/Parsing.scala#L63)  match is tripping up codecov